docs/source/ibis.md

```diff
@@ -0,0 +1,589 @@
+        """Take a single value and return a boolean scalar."""
+        return x > 0
+
+
+t = ibis.memtable({"a": [1, 2, 3]})
+validated_t = t.pipe(ModelWithCustomChecks.validate)
+print(validated_t)
+```
+
+```{testoutput} ibis
+InMemoryTable
+  data:
+    PandasDataFrameProxy:
+         a
+      0  1
+      1  2
+      2  3
+```
+:::
+
+::::
+
+
+For column-level checks, the custom check function should return an Ibis
+table containing a single boolean column or a single boolean scalar.
+
+### DataFrame-level Checks
+
+If you need to validate values on an entire dataframe, you can specify a check
+at the dataframe level. The expected output is an Ibis table containing
+multiple boolean columns, a single boolean column, or a scalar boolean.
+
+::::{tab-set}
+
+:::{tab-item} DataFrameSchema
+
+```{testcode} ibis
+from ibis import _, selectors as s
+
+
+def col1_gt_col2(data: IbisData, col1: str, col2: str) -> ir.Table:
+    """Return a table with a single boolean column."""
+    return data.table.select(data.table[col1] > data.table[col2])
+
+def is_positive_df(data: IbisData) -> ir.Table:
+    """Return a table with multiple boolean columns."""
+    return data.table.select(s.across(s.all(), _ > 0))
+
+def is_positive_element_wise(x: int) -> bool:
+    """Take a single value and return a boolean scalar."""
+    return x > 0
+
+schema_with_df_checks = pa.DataFrameSchema(
+    columns={
+        "a": pa.Column(int),
+        "b": pa.Column(int),
+    },
+    checks=[
+        pa.Check(col1_gt_col2, col1="a", col2="b"),
+        pa.Check(is_positive_df),
+        pa.Check(is_positive_element_wise, element_wise=True),
+    ]
+)
+
+t = ibis.memtable({"a": [2, 3, 4], "b": [1, 2, 3]})
+validated_t = t.pipe(schema_with_df_checks.validate)
+print(validated_t)
+```
+
+```{testoutput} ibis
+InMemoryTable
+  data:
+    PandasDataFrameProxy:
+         a  b
+      0  2  1
+      1  3  2
+      2  4  3
+```
+:::
+
+:::{tab-item} DataFrameModel
+
+```{testcode} ibis
+class ModelWithDFChecks(pa.DataFrameModel):
+    a: int
+    b: int
+
+    @pa.dataframe_check
+    def cola_gt_colb(cls, data: PolarsData) -> pl.LazyFrame:
+        """Return a LazyFrame with a single boolean column."""
+        return data.lazyframe.select(pl.col("a").gt(pl.col("b")))
+
+    @pa.dataframe_check
+    def is_positive_df(cls, data: PolarsData) -> pl.LazyFrame:
+        """Return a LazyFrame with multiple boolean columns."""
+        return data.lazyframe.select(pl.col("*").gt(0))
+
+    @pa.dataframe_check(element_wise=True)
+    def is_positive_element_wise(cls, x: int) -> bool:
+        """Take a single value and return a boolean scalar."""
+        return x > 0
+
+validated_df = lf.collect().pipe(ModelWithDFChecks.validate)
+print(validated_df)
+```
+
+```{testoutput} ibis
+InMemoryTable
+  data:
+    PandasDataFrameProxy:
+         a  b
+      0  2  1
+      1  3  2
+      2  4  3
+```
+:::
+
+::::
+
+
+## Data-level Validation with LazyFrames
```

Comment:
This section hasn't been updated yet, because I haven't yet figured out how to proceed.